jiant/tasks/lib/multirc.py

```diff
@@ -151,22 +152,44 @@ def get_test_examples(self):
 
     def _create_examples(self, lines, set_type):
         examples = []
-        question_id = 0
         for line in lines:
             passage = line["passage"]["text"]
+            passage_id = line["idx"]
             for question_dict in line["passage"]["questions"]:
                 question = question_dict["question"]
+                question_id = question_dict["idx"]
                 for answer_dict in question_dict["answers"]:
+                    answer_id = answer_dict["idx"]
                     answer = answer_dict["text"]
                     examples.append(
                         Example(
-                            guid="%s-%s" % (set_type, line["idx"]),
+                            # NOTE: super_glue_format_preds() is dependent on this guid format.
+                            guid="%s-%s-%s-%s" % (set_type, passage_id, question_id, answer_id),
                             paragraph=passage,
                             question=question,
                             answer=answer,
                             label=answer_dict["label"] if set_type != "test" else self.LABELS[-1],
                             question_id=question_id,
                         )
                     )
-                question_id += 1
         return examples
+
+    @staticmethod
+    def super_glue_format_preds(pred_dict):
+        """Reformat this task's raw predictions to have the structure expected by SuperGLUE."""
+        lines = []
+        # formatting code adapted from: https://github.com/nyu-mll/jiant/blob/
+        # 14fae87d2ebc5a45dbe7254e9007d1a148dd6b18/jiant/evaluate.py#L427
+        par_qst_ans_d = collections.defaultdict(lambda: collections.defaultdict(list))
+        for pred, guid in zip(list(pred_dict["preds"]), list(pred_dict["guids"])):
```

Comment:
The logic in this method is difficult to understand, but this is mostly because the required output format is challenging to work within. To validate that this logic works, I compared the structure structure of some examples produced by this logic to the [sample provided by the leaderboard](https://dl.fbaipublicfiles.com/glue/superglue/baselines/CBoW.zip), and they matched.

Reply:
Thanks for the helpful comments! This looks great. Do you think there will be super_glue_format_preds() will be shared across several tasks?

Reply:
Right now I'm not sure how much overlap there is in the implementation of `super_glue_format_preds()` between tasks (there was a big diff between MultiRC and RTE). I'm planning to move forward with implementing them and maybe we'll see an opportunity to share reusable parts.